pkg/reconciler/pipelinerun/resources/apply_test.go

```diff
@@ -96,6 +96,36 @@ func TestApplyParameters(t *testing.T) {
 			}},
 		},
 		alpha: true,
+	}, {
+		name: "parameter propagation string into finally task",
+		original: v1beta1.PipelineSpec{
+			Finally: []v1beta1.PipelineTask{{
+				TaskSpec: &v1beta1.EmbeddedTask{
+					TaskSpec: v1beta1.TaskSpec{
+						Steps: []v1beta1.Step{{
+							Name:   "step1",
+							Image:  "ubuntu",
+							Script: `#!/usr/bin/env bash\necho "$(params.HELLO)"`,
+						}},
+					},
+				},
+			}},
+		},
+		params: []v1beta1.Param{{Name: "HELLO", Value: *v1beta1.NewStructuredValues("hello param!")}},
+		expected: v1beta1.PipelineSpec{
+			Finally: []v1beta1.PipelineTask{{
+				TaskSpec: &v1beta1.EmbeddedTask{
+					TaskSpec: v1beta1.TaskSpec{
+						Steps: []v1beta1.Step{{
+							Name:   "step1",
+							Image:  "ubuntu",
+							Script: `#!/usr/bin/env bash\necho "hello param!"`,
+						}},
+					},
+				},
+			}},
+		},
+		alpha: true,
```

Comment:
@chitrangpatel please add the other scenarios but testing for finally tasks - like the case where there's a default value

Reply:
Sure, can do.

Reply:
Done!